modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseXPath.java

```diff
@@ -214,22 +214,25 @@ public SynapseXPath(String xpathString) throws JaxenException {
     }
 
     private String getPropertyScope(String xPathString) {
-       String[] args;
-       String xpath = null;
-       String scope = "";
-       if (xPathString.contains("get-property")) {
+        String xpath = null;
+        String scope = "";
+        if (xPathString.contains("get-property")) {
             // extract property args
-            xpath = xPathString.substring(xPathString.indexOf("(") + 1, xPathString.indexOf(")"));
-	    args = xpath.split(",");
-	    if (args != null && args.length == 2) {
-		// remove white space
-		scope = args[0].trim();
-		// remove opening and ending quotes
-		scope = scope.substring(1, scope.length() - 1);
+            Matcher matcher = Pattern.compile("get-property\\(([^)]+)\\)").matcher(xPathString);
```

Comment:
can we use string maniulation insted of regex here
since we are already inside if (xPathString.contains("get-property")) { check and regex is expensive